openpype/hosts/houdini/api/lib.py

```diff
@@ -747,3 +752,41 @@ def get_camera_from_container(container):
 
     assert len(cameras) == 1, "Camera instance must have only one camera"
     return cameras[0]
+
+
+def update_job_var_context():
+    """Validate job path to ensure it matches the settings."""
+
+    project_settings = get_current_project_settings()
+    project_settings = \
+        project_settings["houdini"]["general"]["update_job_var_context"]
+
+    if project_settings["enabled"]:
+
+        # get and resolve job path template
+        job_path_template = project_settings["job_path"]
+        job_path = StringTemplate.format_template(
+            job_path_template, get_current_context_template_data()
+        )
+        job_path = job_path.replace("\\", "/")
+
+        if job_path == "":
+            # Set JOB path to HIP path if JOB path is enabled
+            # and has empty value.
+            job_path = os.environ["HIP"]
+
+        current_job = hou.hscript("echo -n `$JOB`")[0]
+
+        # sync both environment variables.
+        # because when opening new file $JOB is overridden with
+        # the value saved in the HIP file but os.environ["JOB"] is not!
+        os.environ["JOB"] = current_job
+
+        if current_job != job_path:
+            hou.hscript("set JOB=" + job_path)
+            os.environ["JOB"] = job_path
+
+            os.makedirs(job_path, exist_ok=True)
```

Comment:
Do we still support Houdini versions with Py2? If that's the case `exist_ok` argument won't exist in Py2.

Reply:
I think Houdini versions with Py2 still work, and I think there are no yet critical code that has to be py3 specific.
but I didn't try testing with a houdini version with py2. 
so, is this better: 
```python
if not os.path.exists(job_path):
    os.makedirs(job_path)
```

Reply:
```python
if not os.path.exists(job_path):
    os.makedirs(job_path)
```

This would work in Py2 yes. Whether it's better? Purists would say it's worse because it introduces a race condition where just after the `exists` call the folder might have been created already (or removed). See devs having fun discussing the race condition [here](https://stackoverflow.com/a/273227).

That's where they recommend doing e.g.:
```python
import os, errno

try:
    os.makedirs(directory)
except OSError as e:
    if e.errno != errno.EEXIST:
        raise
```